pages/index.js

```diff
@@ -23,12 +36,31 @@ export default function HomePage({
             {unPurchasedItems.length} items to buy{" "}
           </StyledTotalItems>
         </ListHeader>
-        <Form onCreateItem={onCreateItem} />
+
+        {!editItem && (
+          <Form
+            onCreateItem={onCreateItem}
+            editItem={editItem}
+            onToggleIsEditing={toggleIsEditing}
+          />
+        )}
+
+        {editItem && (
+          <Form
+            onEditItem={onEditItem}
+            editItem={editItem}
+            onUpdateItem={updateItem}
+            onToggleIsEditing={toggleIsEditing}
+          />
+        )}
```

Comment:
-  Since you're using `onEditItem` directly in the Form component, there's no need to pass the` onUpdateItem ` function. **Please remove `onUpdateItem` prop and the corresponding function.** 

**Regarding the conditional rendering** 
- The current conditional rendering logic could be simplified.
- Right now, you're rendering two almost identical `<Form />` components based on the value of `editItem`.
- The only difference is whether the form is passed `onCreateItem` or `onEditItem`
- Instead of rendering two separate components based on `editItem`, you can use a single `<Form />` and conditionally pass the correct props. Maybe you can find out how to do this.  🤓
- 💡 _Feel free to keep it as is if it's easier for you to understand, but you can also note it down as a potential refactor for later."_